### PR TITLE
Refactor cleanup decorator

### PR DESCRIPTION
**Pull request by getjump**
#113 Resolved

Also switched from global variable to chat_data context stuff

### CODE REVIEW

**Comment by getjump** [comment]
Added random cure possibility here, since it was missing

File: bot/skills/covid_mode.py [hunk 1]
```diff
@@ -300,21 +309,28 @@ def random_cough(bot: Bot, queue: JobQueue):
...
-                    if lethaled is None:
-                        _db.add_lethality(user['_id'], datetime.now())
-                        bot.restrict_chat_member(get_group_chat_id(), user['_id'],
-                                                 can_add_web_page_previews=False,
-                                                 can_send_media_messages=False,
-                                                 can_send_other_messages=False,
-                                                 can_send_messages=False)
-                        message += f"{full_name} умер от коронавируса, F\n"
+                    _db.add_lethality(user['_id'], datetime.now())
+                    bot.restrict_chat_member(get_group_chat_id(), user['_id'],
```
Comment (egregors) [inline bot/skills/covid_mode.py:328]: ⚠️ this code will lead to failure on python-telegram-bot version 12.4.2 and above. Use `skills.mute.mute_user_for_time` or declaration like: 
```
update.message.reply_text(f"Таймаут для {user.full_name} на {mute_duration}")
        mute_perm = ChatPermissions(
            can_add_web_page_previews=False,
            can_send_media_messages=False,
            can_send_other_messages=False,
            can_send_messages=False
        )
        context.bot.restrict_chat_member(update.effective_chat.id, user.id, mute_perm, until)
```

File: bot/skills/covid_mode.py [hunk 2]
```diff
@@ -370,20 +386,17 @@ def infect_user_masked_condition(user: User, masked_probability: float, unmasked
...
     user: User = update.effective_user
 
     if update.message is not None and update.message.reply_to_message is not None:
         _db.add(update.message.reply_to_message.from_user)
 
     user_to_infect: Optional[User] = None
 
-    if prev_message_user is not None:
+    if PREV_MESSAGE_USER_KEY in context.chat_data.keys():
+        prev_message_user: User = context.chat_data[PREV_MESSAGE_USER_KEY]
```
Comment (egregors) [inline bot/skills/covid_mode.py:398]: uncatched `KeyError`

File: bot/mode.py [hunk 3]
```diff
@@ -167,48 +202,30 @@ def cleanup_decorator(func):
...
-
-            if bot:
-                orig_fn = _hook_message(bot, lambda msg: (
-                    _remove_message_after(msg, queue, seconds)
-                ))
-
-            if bot and update:
+            (bot, message, queue) = _cleanup_args_resolver(*args)
+
+            # No need to do anything else just return original
```
[the remark this answers is not in the document]
Comment (getjump) [reply bot/mode.py:207]: @egregors We have two possible signatures for cleanup decorator: Bot, JobQueue and Update, CallbackContext

**Pull request merged**